tools/taskcluster-worker-runner/protocol/protocol.go

```diff
@@ -11,16 +11,17 @@ type Protocol struct {
 	// transport over which this protocol is running
 	transport Transport
 
-	// current set of agreed capabilities (but call WaitUntilInitialized first
-	// to avoid finding the empty set at startup)
-	Capabilities *Capabilities
-
-	// Capabilities of the remote peer
-	RemoteCapabilities *Capabilities
+	// local and remote capabilities
+	localCapabilities  *capabilities
+	remoteCapabilities *capabilities
 
 	// callbacks per message type
 	callbacks map[string][]MessageCallback
 
+	// tracking for whether Start has been called yet
+	started      bool
+	startedMutex sync.Mutex
```

Comment:
Better to use `RWMutex` to avoid contention.

Reply:
There's only a tiny critical section -- why does it matter?

Reply:
The mutex isn't protecting the start process -- it's just protecting access to the boolean variable.

Reply:
Ah, got it, maybe it would be better to use an atomic variable, but it doesn't matter much.